jl4-lsp/src/LSP/Core/IdeConfiguration.hs

```diff
@@ -86,12 +86,17 @@ isWorkspaceFile uri =
       if isRelative (fromNormalizedFilePath file)
         then return True
         else do
-          IdeConfiguration {..} <- getIdeConfiguration
+          mconfig <- try getIdeConfiguration
+
           let toText = getUri . fromNormalizedUri
-          return $
-            any
-              (\root -> toText root `isPrefixOf` toText (normalizedFilePathToUri file))
-              workspaceFolders
+          pure case mconfig of
+            -- NOTE: if we are not in an editor context, we assume that every file is part
+            -- of our "workspace" as in this case there's not really a concept of a workspace
+            Left (_ :: SomeException) -> True
```

Comment:
I assume we can be more accurate than `SomeException`? I think `safe-exceptions` behaves correctly, but I do feel much better when we are explicitly catching the exception that might occur.